instrumentation/src/com/intellij/rt/coverage/instrumentation/filters/KotlinCoroutinesFilter.java

```diff
@@ -0,0 +1,217 @@
+    return KotlinUtils.isKotlinClass(context)
+        && (name.equals("invokeSuspend") || desc.endsWith("Lkotlin/coroutines/Continuation;)Ljava/lang/Object;"));
+  }
+
+  @Override
+  public void visitLineNumber(int line, Label start) {
+    myHadLineDataBefore = myContext.getLineData(line) != null;
+    super.visitLineNumber(line, start);
+    myLine = line;
+    myHasExecutableCode = false;
+  }
+
+  @Override
+  public void visitMethodInsn(int opcode, String owner, String name, String descriptor, boolean isInterface) {
+    super.visitMethodInsn(opcode, owner, name, descriptor, isInterface);
+    myHasExecutableCode = true;
+
+    boolean getCoroutinesSuspendedVisited = opcode == Opcodes.INVOKESTATIC
+        && owner.equals("kotlin/coroutines/intrinsics/IntrinsicsKt")
+        && name.equals("getCOROUTINE_SUSPENDED")
+        && descriptor.equals("()Ljava/lang/Object;");
+    boolean suspendCallVisited = descriptor.endsWith("Lkotlin/coroutines/Continuation;)Ljava/lang/Object;");
+    if (getCoroutinesSuspendedVisited || suspendCallVisited) {
+      myGetCoroutinesSuspendedVisited |= getCoroutinesSuspendedVisited;
+      mySuspendCallVisited |= suspendCallVisited;
+    } else {
+      myGetCoroutinesSuspendedVisited = false;
+      mySuspendCallVisited = false;
+    }
+  }
+
+  @Override
+  public void visitVarInsn(int opcode, int var) {
+    super.visitVarInsn(opcode, var);
+    myHasExecutableCode = true;
+    if (!myStoreCoroutinesSuspendedVisited && myGetCoroutinesSuspendedVisited && opcode == Opcodes.ASTORE) {
+      myStoreCoroutinesSuspendedVisited = true;
+      myCoroutinesSuspendedIndex = var;
+    }
+    myLoadCoroutinesSuspendedVisited = myStoreCoroutinesSuspendedVisited
+        && opcode == Opcodes.ALOAD
+        && var == myCoroutinesSuspendedIndex;
+  }
+
+  /**
+   * Ignore 'if' if it compares with 'COROUTINE_SUSPENDED' object.
+   * Comparison could be with stored variable or with just returned from 'getCOROUTINE_SUSPENDED' value.
+   */
+  @Override
+  public void visitJumpInsn(int opcode, Label label) {
+    super.visitJumpInsn(opcode, label);
+    myHasExecutableCode = true;
+    boolean compareWithCoroutinesSuspend = myLoadCoroutinesSuspendedVisited || myGetCoroutinesSuspendedVisited;
+    if (compareWithCoroutinesSuspend && mySuspendCallVisited && opcode == Opcodes.IF_ACMPNE) {
+      onIgnoredJump();
+      mySuspendCallVisited = false;
+    }
+    myLoadCoroutinesSuspendedVisited = false;
+  }
+
+  @Override
+  public void visitFieldInsn(int opcode, String owner, String name, String descriptor) {
+    super.visitFieldInsn(opcode, owner, name, descriptor);
+    // ignore return Unit line
+    myHasExecutableCode |= !(owner.equals("kotlin/Unit")
+        && name.equals("INSTANCE")
+        && descriptor.equals("Lkotlin/Unit;"));
+    myStateLabelVisited = opcode == Opcodes.GETFIELD
+        && name.equals("label")
+        && descriptor.equals("I")
+        && owner.startsWith(getClassName());
```

Comment:
Better do not do this yourself, use org.jetbrains.org.objectweb.asm.Type.getObjectType(owner).getClassName()